debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -265,10 +265,17 @@ protected void execute() {
             if (!isRunning()) {
                 return;
             }
-            logger.info("Step 0: disabling autocommit and enabling repeatable read transactions");
+
+            final long snapshotLockTimeout = context.getConnectorConfig().snapshotLockTimeout().getSeconds();
+            logger.info("Step 0: disabling autocommit, enabling repeatable read transactions, and setting lock wait timeout to {}",
+                    snapshotLockTimeout);
             mysql.setAutoCommit(false);
             sql.set("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ");
             mysql.executeWithoutCommitting(sql.get());
+            sql.set("SET SESSION lock_wait_timeout=" + snapshotLockTimeout);
+            mysql.executeWithoutCommitting(sql.get());
+            sql.set("SET SESSION innodb_lock_wait_timeout=" + snapshotLockTimeout);
```

Comment:
Will this cause an issue if using something different than INNODB?

Reply:
It's meaningless if people are extracting MyISAM or MyROCKS tables - but not harmful either. 
```
Your MySQL connection id is 2
Server version: 5.7.28 MySQL Community Server (GPL)
...
mysql> SHOW ENGINES;
+--------------------+---------+----------------------------------------------------------------+--------------+------+------------+
| Engine             | Support | Comment                                                        | Transactions | XA   | Savepoints |
+--------------------+---------+----------------------------------------------------------------+--------------+------+------------+
| InnoDB             | YES     | Supports transactions, row-level locking, and foreign keys     | YES          | YES  | YES        |
| MRG_MYISAM         | YES     | Collection of identical MyISAM tables                          | NO           | NO   | NO         |
| MEMORY             | YES     | Hash based, stored in memory, useful for temporary tables      | NO           | NO   | NO         |
| BLACKHOLE          | YES     | /dev/null storage engine (anything you write to it disappears) | NO           | NO   | NO         |
| MyISAM             | DEFAULT | MyISAM storage engine                                          | NO           | NO   | NO         |
| CSV                | YES     | CSV storage engine                                             | NO           | NO   | NO         |
| ARCHIVE            | YES     | Archive storage engine                                         | NO           | NO   | NO         |
| PERFORMANCE_SCHEMA | YES     | Performance Schema                                             | NO           | NO   | NO         |
| FEDERATED          | NO      | Federated MySQL storage engine                                 | NULL         | NULL | NULL       |
+--------------------+---------+----------------------------------------------------------------+--------------+------+------------+
9 rows in set (0.00 sec)

mysql> SET SESSION innodb_lock_wait_timeout=100;
Query OK, 0 rows affected (0.00 )
```
```
Your MySQL connection id is 8
Server version: 8.0.18 MySQL Community Server - GPL
...
mysql> SHOW ENGINES;
+--------------------+---------+----------------------------------------------------------------+--------------+------+------------+
  1 [mysqld]
| Engine             | Support | Comment                                                        | Transactions | XA   | Savepoints |
+--------------------+---------+----------------------------------------------------------------+--------------+------+------------+
| FEDERATED          | NO      | Federated MySQL storage engine                                 | NULL         | NULL | NULL       |
| MEMORY             | YES     | Hash based, stored in memory, useful for temporary tables      | NO           | NO   | NO         |
| InnoDB             | YES     | Supports transactions, row-level locking, and foreign keys     | YES          | YES  | YES        |
| PERFORMANCE_SCHEMA | YES     | Performance Schema                                             | NO           | NO   | NO         |
| MyISAM             | DEFAULT | MyISAM storage engine                                          | NO           | NO   | NO         |
| MRG_MYISAM         | YES     | Collection of identical MyISAM tables                          | NO           | NO   | NO         |
| BLACKHOLE          | YES     | /dev/null storage engine (anything you write to it disappears) | NO           | NO   | NO         |
| CSV                | YES     | CSV storage engine                                             | NO           | NO   | NO         |
| ARCHIVE            | YES     | Archive storage engine                                         | NO           | NO   | NO         |
+--------------------+---------+----------------------------------------------------------------+--------------+------+------------+
9 rows in set (0.00 sec)

mysql> SET SESSION innodb_lock_wait_timeout=100;
Query OK, 0 rows affected (0.00 sec)
```

Reply:
Ok, thanks. Any chance you could do a test having the `InnoDB` engine not being available at all? I _suppose_ it'll be just ignored in that case, too, but with @jpechane we were contemplating whether this one should be wrapped into a try/catch to swallow that just in case.

Reply:
In MySQL 8 [innodb can no longer be disabled](https://dev.mysql.com/doc/refman/8.0/en/innodb-parameters.html):
> The InnoDB storage engine can no longer be disabled, and the --innodb=OFF and --skip-innodb options are deprecated and have no effect. Their use results in a warning. These options will be removed in a future MySQL release.

_but_ it does throw an error if innodb is disabled in an earlier mysql version. I'll wrap this in a try/catch. Thanks for the feedback!